llvm/lib/SYCLLowerIR/SYCLUtils.cpp

```diff
@@ -43,6 +101,13 @@ void traverseCallgraphUp(llvm::Function *F, CallGraphNodeAction ActionF,
         } else {
           // ... non-call is OK - add using function to the worklist
           if (auto *I = dyn_cast<Instruction>(FCall)) {
+            if (auto *SI = dyn_cast<StoreInst>(I)) {
```

Comment:
as we talked, such filtering within `traverseCallgraphUp` should be opaque, and filtering function should be passed as parameter rather than hardcoded, as there are multiple users of `traverseCallgraphUp`, but we need this invoke_simd behavior only for attribute markup. Somthing like
```cpp
void traverseCallgraphUp(llvm::Function *F, CallGraphNodeAction ActionF,
                         SmallPtrSetImpl<Function *> &FunctionsVisited,
                         bool ErrorOnNonCallUse, std::function<bool(const Instruction *I, const Function *F)> && FuncUseFilter);
```
and the LowerESIMDKernelAttrs pass would invoke the `traverseCallgraphUp` with proper FuncUseFilter argument.